### PR TITLE
SNO-90-added-advanced-search

### PR DESCRIPTION
**Pull request by zoldello**

### CODE REVIEW

**Comment by zoldello** [comment]
@hitz I added a smoke-bdd test for advanced query search, a bunch of unit tests and modified the snovault code a little

File: src/encoded/tests/features/search.feature [hunk 1]
```diff
@@ -80,3 +80,8 @@ Feature: Search
         And I wait for the content to load
         When I fill in "searchOrgan" with "zzz"
         Then I should see 0 elements with the css selector "div.term-list.searchOrgan > li"
+
+    Scenario: Smoke testing advanced query search
+        When I fill in "searchTerm" with "@type:Experiment  date_created:[2015-01-01 TO 2018-12-31]"
```
Comment (hitz) [inline src/encoded/tests/features/search.feature]: wait this should fail?  searchTerm -> advancedQuery?

Comment (zoldello) [reply src/encoded/tests/features/search.feature]: No. On the UI (HTML), the name-attribute on the search element is "searchTerm". So, it does not fail. I did not update the name though as it is beyond the scope of this ticket. On a different topic,  Emma brought up something with advancedQuery, dates and "clear filters" that I need to think about.

**Comment by zoldello** [comment]
Host sno-90.*
  Hostname i-05ad2979f10e22c14.instance.encodedcc.org
  # https://sno-90-97475348f-phil.demo.encodedcc.org
  # ssh ubuntu@i-05ad2979f10e22c14.instance.encodedcc.org

**Comment by caseylitton** [comment]
failing bdd tests was passed on other pr

test_admin_user.py::test_save_a_change_to_an_antibody FAILED

**Pull request merged**